docs/design/task-launch-order-within-job.md

```diff
@@ -0,0 +1,79 @@
+# Task Launch Order Within Job
+
+@[hwdef](https://github.com/hwdef), @[Thor-wl](https://github.com/Thor-wl); July 19th, 2021
+
+## Introduction
+
+This feature provides the ability to customize the order in which tasks are launched.
+
+## Scope
+
+### In Scope
+
+* Reasons why this feature is needed
+* Define the API
+
+### Out of Scope
+
+* Start order of job
+* Task start sequence between multiple jobs
+* Dependency completion state of the task start sequence
+
+## Scenarios
+
+* MPI Job. The worker pods need to be started first and then the master pod can run. Elsewise, the master pod can't setup the ssh tunnel and thus failed, this will add unnecessary waste of resources. In this case, MPI worker pods need to be in the running state, and then the master pod can start. Very similary, for the TensorFlow job, the TF master pod must be started first and then the TF worker pods.
+
+## Scene Comparison
+
+| example   | number of tasks | task dependencies | concurrent execution | Solution Status   | Disadvantages of the current solution |
+| ---------- | -------- | ------------ | -------- | -------------------------------------------------- | -------------------------- |
+| MPI        | 2        | linear dependencies | yes    | 1. Add initcontainer to the task<br/>2.Using user-written check scripts in initcontainer | 1.Increase the cost of use for users<br />2.Still using resources while waiting |
```

Comment:
It'd be better that VCJob provides a convenient image for initContainer and maintains the related codes like MPI-Operator does.

Reply:
There are many limitations to using initcontainer, For example, for windows cluster support, and support for multiple detection methods will not be very convenient. I think add initcontainer for pod is not a good way.